libs/MeshKernel/include/MeshKernel/CurvilinearGrid/CurvilinearGrid.hpp

```diff
@@ -269,6 +254,65 @@ namespace meshkernel
         /// @return a vector of M nodes
         std::vector<Point> GetNodeVectorAtN(UInt n) const { return lin_alg::MatrixColToSTLVector(m_gridNodes, n); }
 
+        /// @brief Get the projection
+        /// @return The curvilinear grid projection
+        [[nodiscard]] const Projection& projection() const { return m_projection; }
+
+        /// @brief Computes the number of nodes
+        [[nodiscard]] UInt GetNumNodes() const { return NumN() * NumM(); }
+
+        /// @brief Computes the number of edges
+        [[nodiscard]] UInt GetNumEdges() const { return NumM() * (NumN() - 1) +
+                                                        (NumM() - 1) * NumN(); }
+
+        /// @brief Computes the number of faces
+        [[nodiscard]] UInt GetNumFaces() const { return (NumN() - 1) * (NumM() - 1); }
+
+        /// @brief Computes the node from an index
+        [[nodiscard]] const Point& Node(const UInt index) const
+        {
+            const UInt n = static_cast<UInt>(std::floor(index / NumM()));
```

Comment:
not sure it will be faster than:

        [[nodiscard]] const Point& Node(const UInt index) const
        {
            const UInt n = index / NumM();
            const UInt m = index % NumM();
            return m_gridNodes(n, m);
        }
the approach above saves some memory